### DISCUSSION TITLE
Does grunt-karma in core libs for Quarto blog need a security update?

### DISCUSSION THREAD

**Discussion by nucleic-acid** [quarto-dev/quarto-cli#7384] [Q&A] [answered]
### Description

I got a security alert from github dependabot in the repository of my personal quarto blog, stating that 
"Prototype pollution vulnerability in karma-runner grunt-karma 4.0.1 via the key variable in grunt-karma.js"

The affected file is _freeze/site_libs/core-js-2.5.3/package.json

Is there some way to update this myself?

**Comment by cscheid** [comment]
We don't put package.json files in your _files directory or use grunt ourselves. That's something you did, I'm pretty sure.

**Reply by cscheid** [reply]
[./_posts/2021-08-22-tea-earl-grey-hot-tidytuesday-2021-week-34/tea-earl-grey-hot-tidytuesday-2021-week-34_files/core-js-2.5.3/package.json](https://github.com/nucleic-acid/jollydatablog/blob/revamp_project/_posts/2021-08-22-tea-earl-grey-hot-tidytuesday-2021-week-34/tea-earl-grey-hot-tidytuesday-2021-week-34_files/core-js-2.5.3/package.json) :)

**Reply by cscheid** [reply]
I don't understand how you generated that file, but it looks like it's from RMarkdown and [this source](https://github.com/nucleic-acid/jollydatablog/blob/revamp_project/_posts/2021-08-22-tea-earl-grey-hot-tidytuesday-2021-week-34/tea-earl-grey-hot-tidytuesday-2021-week-34.Rmd), since I assume you weren't using quarto 2 years ago.

In any case, quarto doesn't put package.json files in your _files directory, ever, unless you have package.json files yourself. In that case, there's nothing we can do.

**Comment by nucleic-acid** [comment]
Thanks for your replies! 

That's odd, I have no idea what that grunt-karma.js or core,js is. I certainly did not add the package.json manually.
That might be some old library from the distill legacy of the blog...
I'll see if I can find what depends on it and remove that.

If I find something of interest, I'll post it here for documentation purpose.

**Comment by jhk0530** [comment, marked as answer]
Hi @nucleic-acid, I had the same issue and figured out what was causing it.

As @cscheid mentioned, it is not from quarto's problem. but from R code chunk.

When you use the `reactable` to render table in quarto. it will include **core-js-2.5.3** which cause these grunt problems.

I'm not sure why it happened, But to solve this issue change `reactable` to other table package. 

For me, I changed `reactable` into `knitr::kable`.

Hope this helpful.

**Reply by jhk0530** [reply]
actually not `reactable`, but `reactR` (see this https://github.com/glin/reactable/issues/245)

**Reply by nucleic-acid** [reply]
Thanks a lot! Never really found the time to look into it deeper.
